synapse/handlers/auth.py

```diff
@@ -1474,6 +1474,8 @@ def _complete_sso_login(
         # URL we redirect users to.
         redirect_url_no_params = client_redirect_url.split("?")[0]
 
+        assert self._sso_redirect_confirm_template is not None
+
```

Comment:
if we're going down this route, it would be good to add some more comments on these assertions explaining why it is `Optional`, and why we don't expect it to be `None` (ie, why is the assertion safe?)